lib/icingadb/icingadb-objects.cpp

```diff
@@ -1733,6 +1741,13 @@ void IcingaDB::SendStartedDowntime(const Downtime::Ptr& downtime)
 		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(downtime->GetTriggerTime() + downtime->GetDuration())));
 	}
 
+	auto duration = downtime->GetDuration();
+	if (downtime->GetFixed() || ! downtime->IsInEffect()) {
+		duration = downtime->GetEndTime() - downtime->GetStartTime();
+	}
+	xAdd.emplace_back("duration");
+	xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(duration)));
```

Comment:
You're sending a downtime start event here, so you know that if it is a flexible downtime, it has triggered. So there should be no need to check if it's effective or was triggered.

Reply:
> I still don't get why it would be a problem if the duration was always the flexible_duration for flexible downtimes. I'd still say it's a 1 hour downtime in a 3 hour window all the time.

Let's do it that way, i.e. always `flexible_duration` for flexible downtimes.